File: Easy/0119_pascals-triangle-ii/pascals-triangle-ii.go
package main

import (
	"fmt"
)

func getRow(rowIndex int) []int {
  row := make([]int, rowIndex + 1)
	row[0] = 1
	for i := 1; i <= rowIndex; i++ {
		for j := i ; j > 0 ; j-- {
			row[j] += row[j - 1]
		}
	}
	return row
}

func main() {
	tests := []int{
		3,
		0,
		1,
	}

	for _, n := range tests {
		fmt.Printf("---\nn: %d\nRow: %v\n", n, getRow(n))
	}
}

File: Easy/0119_pascals-triangle-ii/pascals-triangle-ii.py
#!/bin/python3

from typing import List

class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        row: List[int] = [0] * (rowIndex + 1)
        row[0] = 1
        for i in range(1, rowIndex + 1):
            for j in range(i, 0, -1):
                row[j] += row[j - 1]
        return row

def main():
    tests: List[int] = [
        3,
        0,
        1,
    ]

    for n in tests:
        print(f"---\nn: {n}\nRow: {Solution().getRow(n)}")

if __name__ == "__main__":
    main()
